### PR TITLE
feature-KH2WEB-2/add-header 

### PR DESCRIPTION
**Pull request by maria-khudolii**
- Added header
- logo, navigation, login and search icon
Screenshot_6

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,6 +1,9 @@
+import Header from './Header';
+
 function App() {
   return (
     <div>
+      <Header></Header>
```
Comment (dima-diachkov) [inline src/App.tsx:6]: if there is nothing in the tag, then 
`<Header />`
 it is enough

File: src/Header/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      </Toolbar>
+    );
+  };
+
+  return (
+    <header>
+      <AppBar>{displayDesktop()}</AppBar>
+    </header>
+  );
+}
```
Comment (dima-diachkov) [inline src/Header/index.tsx:45]: Doesn't read perfectly for me. at first glance, it looks like two returns. then you realize that the first return from displayDesktop.  it's probably better to take this constant out of Header. Well ideally create a separate component.

File: src/App.tsx [hunk 3]
```diff
@@ -1,6 +1,9 @@
...
       <h1>
         What to watch
       </h1>
```
Comment (kerrimov) [inline src/App.tsx:9]: should be removed as Header was added

File: src/Header/index.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import "./style.scss";
+import logo from "./logo.svg";
+
+export default function Header() {
+  const displayDesktop = () => {
+    return (
+      <Toolbar className="header">
+        <div className="menu-container">
+          <a href="/">
+            <img className="logo" src={logo} alt="" />
```
Comment (kerrimov) [inline src/Header/index.tsx:13]: please add alt

File: src/Header/index.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            </li>
+            <li className="menu-item">
+              <a href="/">TV Shows</a>
+            </li>
+            <li className="menu-item">
+              <a href="/">People</a>
+            </li>
+            <li className="menu-item">
+              <a href="/">More</a>
+            </li>
```
Comment (kerrimov) [inline src/Header/index.tsx:27]: please move to separete component and use iterator for it

File: src/Header/style.scss [hunk 6]
```diff
@@ -0,0 +1,44 @@
+.header {
+  background: #297f8e;
```
Comment (kerrimov) [inline src/Header/style.scss:2]: should be moved to variables

File: src/App.tsx (hunk 3 above)
Comment (maria-khudolii) [reply src/App.tsx]: Removed

File: src/Header/index.tsx (hunk 2 above)
Comment (maria-khudolii) [reply src/Header/index.tsx]: Changed the structure completely

File: src/Header/index.tsx (hunk 4 above)
Comment (maria-khudolii) [reply src/Header/index.tsx]: Added

File: src/Header/index.tsx (hunk 5 above)
Comment (maria-khudolii) [reply src/Header/index.tsx]: Separate component done, but failed to do the iterator

File: src/Header/style.scss (hunk 6 above)
Comment (maria-khudolii) [reply src/Header/style.scss]: done

File: src/components/Header/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import AppBar from "@mui/material/AppBar";
+import Toolbar from "@mui/material/Toolbar";
+import Container from "@mui/material/Container";
```
Comment (dima-diachkov) [inline src/components/Header/Header.tsx:5]: these imports can be combined like `import { AppBar, ToolBar, Container } from "@mui/material"`

File: src/pages/Homepage.tsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+
+const wrapperMain = {
+  paddingTop: "120px",
+};
+
+const Homepage = () => {
+  return (
```
Comment (dima-diachkov) [inline src/pages/Homepage.tsx:8]: perhaps the return is not needed here and the syntax should be `const Homepage = () => ()`

File: src/pages/More.tsx [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+const wrapperMain = {
+  paddingTop: "120px",
+};
+
+const More = () => {
+  return (
```
Comment (dima-diachkov) [inline src/pages/More.tsx:7]: the same

**Comment by maria-khudolii** [comment]
Added search field to header
image

**Comment by maria-khudolii** [comment]
Search field added to header 
image

File: src/App.test.tsx [hunk 10]
```diff
@@ -1,9 +1,10 @@
...
 
-test("renders learn react link", () => {
-  render(<App />);
-  const linkElement = screen.getByText(/What to watch/i);
-  expect(linkElement).toBeInTheDocument();
+test("render without crashing", () => {
+  const div = document.createElement("div");
+  const root = createRoot(div);
+  root.render(<App />);
+  root.unmount();
```
Comment (kerrimov) [inline src/App.test.tsx:9]: undo these changes @dima-diachkov has fixed the test in his PR. Will be merged as soon as comments are fixed

File: src/App.tsx [hunk 11]
```diff
@@ -1,10 +1,30 @@
...
+      <ThemeProvider theme={customTheme}>
+        <Header />
+        <Routes>
+          <Route path="/" element={<Homepage />} />
+          <Route path="/movies" element={<MoviesPage />} />
+          <Route path="/tvshows" element={<TVShows />} />
+          <Route path="/people" element={<People />} />
+          <Route path="/more" element={<More />} />
+          <Route path="*" element={<NotFoundPage />} />
+        </Routes>
```
Comment (kerrimov) [inline src/App.tsx:25]: create component Router.tsx and move

**Review by kerrimov — COMMENTED** [review]
move all files from pages to components

File: src/App.tsx (hunk 11 above)
Comment (maria-khudolii) [reply src/App.tsx]: Done

File: src/App.test.tsx (hunk 10 above)
Comment (maria-khudolii) [reply src/App.test.tsx]: pulled the master branch

File: yarn.lock [hunk 12]
```diff
@@ -1578,10 +1578,23 @@
...
+=======
+>>>>>>> 0474db1 (header added)
+"@mui/icons-material@^5.8.0":
+  version "5.8.0"
+  resolved "https://registry.yarnpkg.com/@mui/icons-material/-/icons-material-5.8.0.tgz#c85768434bda885c64a8e4b9e222e73912244570"
+  integrity sha512-ScwLxa0q5VYV70Jfc60V/9VD0b9SvIeZ0Jddx2Dt2pBUFFO9vKdrbt9LYiT+4p21Au5NdYIb2XSHj46CLN1v3g==
+<<<<<<< HEAD
+>>>>>>> 0474db1 (header added)
+=======
+>>>>>>> 0474db1 (header added)
```
Comment (kerrimov) [inline yarn.lock:1597]: please resolve conflicts

File: src/components/Router/Router.tsx [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+import { PageNotFound } from "../PageNotFound";
+
+export const Router = () => (
+  <Routes>
+    <Route path="/" element={<PageHome />} />
+    <Route path="/movies" element={< PageMovies />} />
+    <Route path="/tvshows" element={< PageTVShows />} />
+    <Route path="/people" element={< PagePeople />} />
+    <Route path="/more" element={< PageMore />} />
+    <Route path="*" element={< PageNotFound />} />
```
Comment (kerrimov) [inline src/components/Router/Router.tsx:17]: no need to add "Page" before all components. Just use a more convenient name

File: src/components/PageTVShows/PageTVShows.tsx [hunk 14]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+const wrapperMain = {
+  padding: "120px 50px 50px 50px",
+};
```
Comment (kerrimov) [inline src/components/PageTVShows/PageTVShows.tsx:5]: this code was duplicated several times. Please create Container component in the shared folder and use it everywhere

File: src/components/PageNotFound/PageNotFound.tsx [hunk 15]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+const wrapperMain = {
+  padding: "120px 50px 50px 50px",
+};
+
+export const PageNotFound = () => (
+    <div style={wrapperMain}>
+      This page does not exist. Go <a href="/">home</a>
```
Comment (kerrimov) [inline src/components/PageNotFound/PageNotFound.tsx:9]: why `a` tag,  not `Link`?

**Review by kerrimov — COMMENTED** [review]
please squash your commits

File: src/components/PageTVShows/PageTVShows.tsx (hunk 14 above)
Comment (maria-khudolii) [reply src/components/PageTVShows/PageTVShows.tsx:5]: Did this so that they stay together in the folder.

File: src/components/PageNotFound/PageNotFound.tsx (hunk 15 above)
Comment (maria-khudolii) [reply src/components/PageNotFound/PageNotFound.tsx:9]: Done

File: src/components/PageTVShows/PageTVShows.tsx (hunk 14 above)
Comment (maria-khudolii) [reply src/components/PageTVShows/PageTVShows.tsx:5]: Done

File: src/components/Router/Router.tsx (hunk 13 above)
Comment (maria-khudolii) [reply src/components/Router/Router.tsx:17]: Did this so that they stayed together in the components folder.
Made the subfolder Pages in the components folder, moved them there. Hope that's ok

File: yarn.lock (hunk 12 above)
Comment (maria-khudolii) [reply yarn.lock:1597]: done

File: src/components/Router/Router.tsx (hunk 13 above)
Comment (kerrimov) [reply src/components/Router/Router.tsx]: just leave as before, but without 'page'

File: src/components/PageTVShows/PageTVShows.tsx (hunk 14 above)
Comment (kerrimov) [reply src/components/PageTVShows/PageTVShows.tsx]: it should be HOC component, not an object

File: src/components/Router/Router.tsx (hunk 13 above)
Comment (maria-khudolii) [reply src/components/Router/Router.tsx]: done

File: src/components/Home/Home.tsx [hunk 16]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import "../../components/Header/style.scss";
+
+export const Home = () => (
+  <div className="wrapper">
```
Comment (kerrimov) [inline src/components/Home/Home.tsx:5]: you can't import styles from Header to Home component etc.

Comment (maria-khudolii) [reply src/components/Home/Home.tsx]: Suspected that. But where should the styles be placed? 
Deleted the import and the wrapper class. It had just paddings in it.

Comment (kerrimov) [reply src/components/Home/Home.tsx]: In the shared folder. Discussed on the daily

Comment (maria-khudolii) [reply src/components/Home/Home.tsx]: done

File: src/components/Header/Header.tsx [hunk 17]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { AppBar, Toolbar } from "@mui/material";
+import { Logo } from "./Logo";
+import { Navigation } from "./Navigation";
+import { LoginLink } from "./LoginLink";
+import "./headerStyle.scss";
```
Comment (kerrimov) [inline src/components/Header/Header.tsx:6]: scss file should be named as component

**Review by kerrimov — COMMENTED** [review]
All changes related to search feature should be removed from PR

File: src/components/Header/Header.tsx (hunk 17 above)
Comment (maria-khudolii) [reply src/components/Header/Header.tsx]: renamed

**Comment by kerrimov** [comment]
Please also squash commits

File: src/components/Header/Header.tsx (hunk 17 above)
Comment (kerrimov) [reply src/components/Header/Header.tsx]: file should be renamed too

File: src/shared/SecondaryButton/SecondaryButton.tsx [hunk 18]
```diff
@@ -14,8 +14,8 @@ const style = {
   }
 };
 
-export const SecondaryButton: React.FC<Props> = ({ children }: Props) => (
+export const SecondaryButton: React.FC<Props> = (props: Props) => (
```
[the remark this answers is not in the document]
Comment (kerrimov) [reply src/shared/SecondaryButton/SecondaryButton.tsx:17]: destructuring is a good suit here

File: src/components/Header/Header.tsx (hunk 17 above)
Comment (maria-khudolii) [reply src/components/Header/Header.tsx]: done

**Review by kerrimov — APPROVED** [review]
